apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -540,6 +544,26 @@ func (i *ClowdApp) GetClowdSAName() string {
 	return fmt.Sprintf("%s-app", i.GetClowdName())
 }
 
+func (i *ClowdApp) RunPreHook() bool {
+	// is there a better way to find out if the app has a job?
+	if i.Spec.PreHookJob.Name == "" || i.Annotations["clowder/pre-hook-status"] == "pending" {
+		return false
+	} else if i.Annotations["clowder/pre-hook-generation"] == "" {
```

Comment:
I'm still not 100% bought on this. I think it would perhaps be better to shift this state store onto the Job itself. The other problem you have is that it looks as if we are replacing the same job again and again. I'm not sure if this is the best idea, perhaps having a job per generation would be better, if jobs eventually get cleaned up anyway. That way the job name would be easy to search for as it's a contraction of the apps name and its generation. If you wanted to be more granular and only change when the image changed you could use the image tag as a name component. Of course this brings up the question of what happens if two spec changes are written in quick succession, triggering two jobs, but then I'm not 100% sure that this is solved even as we stand currently.